File: api/src/main/java/gg/mineral/api/math/Location.java
package gg.mineral.api.math;

import gg.mineral.api.world.World;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@AllArgsConstructor
@Getter
public class Location {
    private final World world;
    private final double x, y, z;
    private float yaw, pitch;
}

File: api/src/main/kotlin/gg/mineral/api/math/Location.kt
package gg.mineral.api.math

import gg.mineral.api.world.World

class Location(
    val world: World,
    val x: Double,
    val y: Double,
    val z: Double,
    val yaw: Float = 0.0f,
    val pitch: Float = 0.0f
)

